### PR TITLE
Add testing for scalar functions with float

### PR DESCRIPTION
**Pull request by akshatnair**
Testing: unit tests

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java [hunk 1]
```diff
@@ -143,18 +146,7 @@ public void testCeil()
         assertFunction("ceil(FLOAT '123.45')", DOUBLE, 124.0);
         assertFunction("ceil(FLOAT '-123.45')", DOUBLE, -123.0);
         assertFunction("ceil(CAST(NULL as DOUBLE))", DOUBLE, null);
-        assertFunction("ceiling(12300000000)", BIGINT, 12300000000L);
```
Comment (akshatnair) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java:146]: duplicate tests

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java [hunk 2]
```diff
@@ -70,7 +72,7 @@ public void testAcos()
     {
         for (double doubleValue : DOUBLE_VALUES) {
             assertFunction("acos(" + doubleValue + ")", DOUBLE, Math.acos(doubleValue));
-            assertFunction("acos(FLOAT '" + (float) doubleValue + "')", DOUBLE, Math.acos((float) doubleValue));
+            assertFunction("acos(FLOAT '" + doubleValue + "')", DOUBLE, Math.acos((float) doubleValue));
```
Comment (cawallin) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java:75]: Are you going to change these? I'd imagine that trig functions would be very susceptible to double/float changes

Comment (akshatnair) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java:75]: You mean add acos that accepts float? We need to have a discussion about this on Monday. I briefly spoke to @maciejgrzybek about this. From what I see on Postgres it only accepts Double precision and returns Double precision. Need to check what is the standard for this is, last time when we did something similar we followed the types for functions supported by Postgres

Comment (maciejgrzybek) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java:75]: This change introduces inconsistency. You build FLOAT value in SQL based on `Double.toString()` while comparing result to `acos(float)` when float is casted from double value by Java casting. I'd keep `(float) doubleValue` in FLOAT c-tor for consistency. This test should not check constructing FLOAT out of Double.

Comment (akshatnair) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java:75]: Okay there were other places where it was just taking doubleValue in existing tests that you added so I wasn't sure what your intent was. Lets chat and decide which approach is better.

**Comment by akshatnair** [comment]
closing this, opened a new PR #236

**Pull request closed without merging**